editor.planx.uk/src/pages/Preview/Node.tsx

```diff
@@ -41,47 +67,49 @@ export type HandleSubmit = (userData?: Store.UserData | Event) => void;
 interface Props {
   handleSubmit: HandleSubmit;
   node: Store.Node;
-  data?: any;
+  data?: Store.Node["data"];
 }
 
 const Node: React.FC<Props> = (props) => {
-  const [
-    childNodesOf,
-    isFinalCard,
-    resetPreview,
-    cachedBreadcrumbs,
-  ] = useStore((state) => [
-    state.childNodesOf,
-    state.isFinalCard(),
-    state.resetPreview,
-    state.cachedBreadcrumbs,
-  ]);
+  const [childNodesOf, isFinalCard, resetPreview, cachedBreadcrumbs] = useStore(
+    (state) => [
+      state.childNodesOf,
+      state.isFinalCard(),
+      state.resetPreview,
+      state.cachedBreadcrumbs,
+    ],
+  );
 
   const handleSubmit = isFinalCard ? undefined : props.handleSubmit;
 
   const nodeId = props.node.id;
   const previouslySubmittedData =
     nodeId && cachedBreadcrumbs ? cachedBreadcrumbs[nodeId] : undefined;
-  const allProps = {
+
+  const getComponentProps = <T extends object>() => ({
     id: nodeId,
-    ...props.node.data,
     previouslySubmittedData,
     resetPreview,
     handleSubmit,
-  };
+    ...(props.node.data as T),
+  });
```

Comment:
Feels like a good direction for moving towards a better typed graph & node types :+1: